test/test_unary_ufuncs.py

```diff
@@ -33,6 +273,149 @@ def test_unsupported_dtypes(self, device, dtype, op):
         with self.assertRaises(RuntimeError):
             op(t)
 
+    @dtypes(*floating_types_and(torch.bfloat16, torch.half))
+    @ops((_fn for _fn in unary_ufuncs if _fn.domain is not None))
+    def test_float_domains(self, device, dtype, op):
+        if not op.supports_dtype(dtype, torch.device(device).type):
+            raise unittest.SkipTest('unsupported dtype')
+
+        eps = (1e-1, 1, 2, 10, 20, 50, 100)
+
+        # Adds smaller values except on bfloat16, since these small differences
+        #   are often too small to represent in bfloat16
+        if dtype is not torch.bfloat16:
+            eps = eps + (1e-5, 1e-3)
+
+        for base, _fn in ((op.domain[0], operator.sub), (op.domain[1], operator.add)):
+            for epsilon in eps:
+                v = _fn(base, epsilon)
+                t = torch.tensor(v, device=device, dtype=dtype)
+                result = op(t)
+                self.assertEqual(result.item(), float('nan'),
+                                 msg="input:{0}".format(t.item()))
+
+    # Tests that fn == method == inplace == jit on a simple single tensor input
+    @ops(unary_ufuncs)
+    def test_variant_consistency(self, device, dtype, op):
```

Comment:
we should test that inplace variant errors out with broadcasted input, I don't know if this test is a good place for this. 
There are some functions that currently don't test for memory overlap and thus don't error out, and some are even well-defined, like fill_, but for most erroring out is the correct behavior.

Reply:
Added a TODO for this